index.js

```diff
@@ -1,43 +1,52 @@
 const http = require('http');
 const events = require('events');
-const {EventEmitter} = events;
+const { EventEmitter } = events;
 const express = require('express');
 const ws = require('ws');
 const engine = require('./build/Release/qr.node');
 
+async function initQr(qrEmitter, locomotionEmitter) {
```

Comment:
If the function is already `async` then there is no need to also wrap in a `Promise`.

This function looks like it should be `async` and throw rather than returning a `Promise` in the body.

Reply:
I'm not sure how to `throw` just an error code like I am currently doing with the promise rejection.

I could do `throw new Error(resultCode)` and then in the code checking for the specific error in metachromium (https://github.com/webaverse/metachromium/pull/4/files#diff-168726dbe96b3ce427e7fedce31bb0bcR67) do something like `if (error.toString() === 'Error: 100')` instead but it feels a bit more hacky so not sure what the best way to do it is!

Reply:
`async` functions will reject the `Promise` as the result of a `throw`.

Reply:
But it will reject the Promise as an actual `Error` object rather than just the `resultCode` I pass into the throw I think?

Reply:
Couldn't it `throw` the number?

Reply:
Ah, TIL you don't have to `throw` an actual `Error`.

Thanks! Will push that now :)

Reply:
I just pushed this change!